### ISSUE TITLE
ares_gethostbyname.c no longer C89 compliant in 1.15 on Windows, won't compile

### ISSUE THREAD

**Issue by jamadden** [c-ares/c-ares#246]
I need to build c-ares using MSVC compiler used for Python 2.7 (which calls itself version 9.0). Unfortunately, [that breaks in 1.15](https://ci.appveyor.com/project/denik/gevent/builds/23978005/job/xoe6boi4k1m1x8tj#L407):
```
C:\Users\appveyor\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -DFD_SETSIZE=1024 -D_WIN32=1 -DCARES_STATICLIB= -DCARES_EMBED=1 -Ibuild\temp.win-amd64-2.7\Release\c-ares -Isrc/gevent/resolver -IC:\projects\gevent\deps\c-ares -IC:\Python27-x64\include -IC:\Python27-x64\PC /Tcdeps/c-ares\ares_gethostbyname.c /Fobuild\temp.win-amd64-2.7\Release\deps/c-ares\ares_gethostbyname.obj
 ares_gethostbyname.c
deps/c-ares\ares_gethostbyname.c(354) : error C2143: syntax error : missing ';' before 'type'
deps/c-ares\ares_gethostbyname.c(355) : error C2275: 'win_platform' : illegal use of this type as an expression
            c:\projects\gevent\deps\c-ares\ares_platform.h(31) : see declaration of 'win_platform'
deps/c-ares\ares_gethostbyname.c(355) : error C2146: syntax error : missing ';' before identifier 'platform'
deps/c-ares\ares_gethostbyname.c(355) : error C2065: 'platform' : undeclared identifier
deps/c-ares\ares_gethostbyname.c(357) : error C2065: 'PATH_HOSTS' : undeclared identifier
deps/c-ares\ares_gethostbyname.c(357) : error C2109: subscript requires array or pointer type
deps/c-ares\ares_gethostbyname.c(359) : error C2065: 'platform' : undeclared identifier
deps/c-ares\ares_gethostbyname.c(361) : error C2065: 'platform' : undeclared identifier
deps/c-ares\ares_gethostbyname.c(371) : error C2065: 'PATH_HOSTS' : undeclared identifier
    deps/c-ares\ares_gethostbyname.c(371) : warning C4047: 'function' : 'LPSTR' differs in levels of indirection from 'int'
    deps/c-ares\ares_gethostbyname.c(371) : warning C4024: 'ExpandEnvironmentStringsA' : different types for formal and actual parameter 2
    deps/c-ares\ares_gethostbyname.c(375) : error C2065: 'platform' : undeclared identifier
    deps/c-ares\ares_gethostbyname.c(376) : error C2065: 'PATH_HOSTS' : undeclared identifier
    deps/c-ares\ares_gethostbyname.c(376) : warning C4047: 'function' : 'LPSTR' differs in levels of indirection from 'int'
    deps/c-ares\ares_gethostbyname.c(376) : warning C4024: 'GetWindowsDirectoryA' : different types for formal and actual parameter 1
    deps/c-ares\ares_gethostbyname.c(380) : error C2065: 'PATH_HOSTS' : undeclared identifier
    deps/c-ares\ares_gethostbyname.c(380) : warning C4047: 'function' : 'char *' differs in levels of indirection from 'int'
    deps/c-ares\ares_gethostbyname.c(380) : warning C4024: 'strcat' : different types for formal and actual parameter 1
    deps/c-ares\ares_gethostbyname.c(390) : error C2065: 'PATH_HOSTS' : undeclared identifier
    deps/c-ares\ares_gethostbyname.c(390) : warning C4047: 'function' : 'const char *' differs in levels of indirection from 'int'
    deps/c-ares\ares_gethostbyname.c(390) : warning C4024: 'fopen' : different types for formal and actual parameter 1
    error: command 'C:\\Users\\appveyor\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual C++ for Python\\9.0\\VC\\Bin\\amd64\\cl.exe' failed with exit status 2
```

Here's the relevant portion of the diff from 1.14 to 1.15:
```diff
@@ -339,6 +346,10 @@ static int file_lookup(const char *name, int family, struct hostent **host)
   int status;
   int error;

+  /* Per RFC 7686, reject queries for ".onion" domain names with NXDOMAIN. */
+  if (ares__is_onion_domain(name))
+    return ARES_ENOTFOUND;
+
 #ifdef WIN32
   char PATH_HOSTS[MAX_PATH];
   win_platform platform;
```

If I remember my ancient C correctly, you can't have any code before a variable declaration, so this `if/return` before the declaration of `PATH_HOSTS` and `platform` seems to be the problem.

**Comment by jamadden** [comment]
FWIW, simply [moving the `if` down](https://github.com/gevent/gevent/commit/c15c219e97fee7b07060e24451e0f8ebb68ee43f#diff-29c7ee201362685f1526b869aaa56a47) below the entire `#ifdef` block (`WIN32` and `WATT32`) solves the problem for me. The cost would seem to be that those two blocks will look for the host file, even if we're going to immediately fail the lookup.

**Comment by bradh352** [comment]
fixed by 9f1fdbf

**Comment by jamadden** [comment]
Thanks!
